src/commands.rs

```diff
@@ -326,8 +326,22 @@ fn decompress_file(
         Gzip | Bzip | Lzma | Zstd => {
             reader = chain_reader_decoder(&formats[0].compression_formats[0], reader)?;
 
-            // TODO: improve error treatment
-            let mut writer = fs::File::create(&output_path)?;
+            let mut writer = match fs::OpenOptions::new().write(true).create_new(true).open(&output_path) {
+                Ok(w) => w,
+                Err(e) if e.kind() == io::ErrorKind::AlreadyExists => {
+                    if utils::user_wants_to_overwrite(&output_path, question_policy)? {
+                        if output_path.is_dir() {
+                            // We can't just use `fs::File::create(&output_path)` because it would return io::ErrorKind::IsADirectory
+                            // ToDo: Maybe we should emphasise that `output_path` is a directory and everything inside it will be gone?
+                            fs::remove_dir_all(&output_path)?;
+                        }
+                        fs::File::create(&output_path)?
+                    } else {
+                        return Ok(());
+                    }
+                }
+                Err(e) => return Err(Error::from(e)),
+            };
```

Comment:
Maybe if you move this to utils.rs, you can reuse the same code for the .zip and .tar parts?

Would that be possible? something like `create_file_checking_overwrites()` which receives one AsRef<Path> and tries to open with those OpenOptions configs you used, and returns `Result<Option<File>>`, with `Ok(None)` in case the used does not want to overwrite.

Reply:
Yeah that's a good point

Reply:
I don't think this is really doable after all because the archives don't want the file to exist there at all. Good news is I caught a bug because of this.
I did extract the function anyway because I think it's a bit cleaner but I didn't end up using it in .zip and .tar
It *could* be used if I were to add some flags determining if the file should be created or not but I don't think it's worth it, what do you think?

Reply:
I don't think that's worth too, but by how that function looked, I think it's better now on utils.rs :D .